.scripts/test_all.sh

```diff
@@ -26,6 +26,30 @@ else
   go get -u github.com/hashicorp/consul/...
 fi
 
+install_etcd_pkgs() {
+  go get -u github.com/coreos/etcd
+  go get -u github.com/gogo/protobuf/gogoproto
+  go get -u google.golang.org/grpc
+  go get -u github.com/coreos/go-semver
+  go get -u github.com/ugorji/go/codec
+  go get -u github.com/mwitkow/go-etcd-harness
+}
+
+# The etcd 3.3.x package only supports Go1.9+ and therefore
+# we will only run the corresponding tests on these versions.
+if [[ "$goversion" == *"go1.5"* ]] ||
+   [[ "$goversion" == *"go1.6"* ]] ||
+   [[ "$goversion" == *"go1.7"* ]] ||
+   [[ "$goversion" == *"go1.8"* ]]; then
+  echo "Skipping tests requiring etcd3locker, which is not supported on $goversion"
+  packages=$(echo "$packages" | sed '/etcd3locker/d')
+else
+  # Install the etcd packages which are not vendored.
+  install_etcd_pkgs
+  # use release 3.3 as master branches are not stable for etcd
+  (cd ../../coreos/etcd && git fetch origin && git checkout release-3.3)
```

Comment:
Nice workaround to get a stable release of etcd but this leaves consumers of the tusd library having to do the same workaround.
What about using etcd v3 directly using gopkg.in? If you use the `gopkg.in/etcd-io/etcd.v3` package, it will use the latest v3 release of etcd (v.3.3.9 currently). If you are not familiar, gopkg.in (https://gopkg.in) is a nice service to pinpoint Go packages using git tags.

Reply:
I'm having issues with using `gopkg.in/etcd-io/etcd.v3` 😢 
```
[08:43:22] etcd3locker(anders/etcd3locker *)$ go get gopkg.in/coreos/etcd.v3
# github.com/coreos/etcd/etcdmain
../../../coreos/etcd/etcdmain/config.go:286:15: cannot use "github.com/coreos/etcd/vendor/go.uber.org/zap".String("path", cfg.configFile) (type "github.com/coreos/etcd/vendor/go.uber.org/zap/zapcore".Field) as type "go.etcd.io/etcd/vendor/go.uber.org/zap/zapcore".Field in argument to lg.Info
../../../coreos/etcd/etcdmain/etcd.go:66:47: cannot use "github.com/coreos/etcd/vendor/go.uber.org/zap".Error(err) (type "github.com/coreos/etcd/vendor/go.uber.org/zap/zapcore".Field) as type "go.etcd.io/etcd/vendor/go.uber.org/zap/zapcore".Field in argument to lg.Warn
../../../coreos/etcd/etcdmain/etcd.go:73:59: cannot use "github.com/coreos/etcd/vendor/go.uber.org/zap".Error(err) (type "github.com/coreos/etcd/vendor/go.uber.org/zap/zapcore".Field) as type "go.etcd.io/etcd/vendor/go.uber.org/zap/zapcore".Field in argument to lg.Warn
../../../coreos/etcd/etcdmain/etcd.go:104:15: cannot use "github.com/coreos/etcd/vendor/go.uber.org/zap".String("host", defaultHost) (type "github.com/coreos/etcd/vendor/go.uber.org/zap/zapcore".Field) as type "go.etcd.io/etcd/vendor/go.uber.org/zap/zapcore".Field in argument to lg.Info
../../../coreos/etcd/etcdmain/etcd.go:112:54: cannot use "github.com/coreos/etcd/vendor/go.uber.org/zap".Error(dhErr) (type "github.com/coreos/etcd/vendor/go.uber.org/zap/zapcore".Field) as type "go.etcd.io/etcd/vendor/go.uber.org/zap/zapcore".Field in argument to lg.Info
../../../coreos/etcd/etcdmain/etcd.go:123:15: cannot use "github.com/coreos/etcd/vendor/go.uber.org/zap".String("data-dir", cfg.ec.Dir) (type "github.com/coreos/etcd/vendor/go.uber.org/zap/zapcore".Field) as type "go.etcd.io/etcd/vendor/go.uber.org/zap/zapcore".Field in argument to lg.Warn
../../../coreos/etcd/etcdmain/etcd.go:133:48: cannot use cfg.ec.GetLogger() (type *"go.etcd.io/etcd/vendor/go.uber.org/zap".Logger) as type *"github.com/coreos/etcd/vendor/go.uber.org/zap".Logger in argument to identifyDataDirOrDie
../../../coreos/etcd/etcdmain/etcd.go:138:15: cannot use "github.com/coreos/etcd/vendor/go.uber.org/zap".String("data-dir", cfg.ec.Dir) (type "github.com/coreos/etcd/vendor/go.uber.org/zap/zapcore".Field) as type "go.etcd.io/etcd/vendor/go.uber.org/zap/zapcore".Field in argument to lg.Info
../../../coreos/etcd/etcdmain/etcd.go:139:15: cannot use "github.com/coreos/etcd/vendor/go.uber.org/zap".String("dir-type", string(which)) (type "github.com/coreos/etcd/vendor/go.uber.org/zap/zapcore".Field) as type "go.etcd.io/etcd/vendor/go.uber.org/zap/zapcore".Field in argument to lg.Info
../../../coreos/etcd/etcdmain/etcd.go:153:16: cannot use "github.com/coreos/etcd/vendor/go.uber.org/zap".String("dir-type", string(which)) (type "github.com/coreos/etcd/vendor/go.uber.org/zap/zapcore".Field) as type "go.etcd.io/etcd/vendor/go.uber.org/zap/zapcore".Field in argument to lg.Panic
../../../coreos/etcd/etcdmain/etcd.go:153:16: too many errors
```

Reply:
Ok, I also tried to getting gopkg.in and etcd to work together but ran into similar problems. Let's simply stick with plain `go get` for now.
By the way, as it seems `go.etcd.io/etcd/clientv3` is now the recommended import path (see https://github.com/etcd-io/etcd/blob/master/clientv3/README.md#install). Would you mind switching to that?

Reply:
All done